pipeline/metadata/ip_metadata_chooser.py

```diff
@@ -4,45 +4,49 @@
 
 from pipeline.metadata import caida_ip_metadata
 from pipeline.metadata import dbip
-from pipeline.metadata.flatten import Row
+from pipeline.metadata.schema import IpMetadata
 
 
 # pylint: disable=too-many-instance-attributes
 class IpMetadataChooser():
   """Business logic for selecting IP metadata values from a number of sources"""
 
-  def __init__(self, caida_db: caida_ip_metadata.CaidaIpMetadata,
+  def __init__(self, date: datetime.date,
+               caida_db: caida_ip_metadata.CaidaIpMetadata,
                dbip_db: dbip.DbipMetadata) -> None:
     """Store all metadata sources for future querying
 
     Args:
       caida: CAIDA database
       dbip: dbip database
     """
+    self.date = date
     self.caida = caida_db
     self.dbip = dbip_db
 
-  def get_metadata(self, ip: str) -> Row:
+  def get_metadata(self, ip: str) -> IpMetadata:
     """Pick which metadata values to return for an IP from our sources."""
     try:
       (netblock, asn, as_name, as_full_name, as_type,
        country) = self.caida.lookup(ip)
-      metadata_values = {
-          'netblock': netblock,
-          'asn': asn,
-          'as_name': as_name,
-          'as_full_name': as_full_name,
-          'as_class': as_type,
-          'country': country,
-      }
+      metadata_values = IpMetadata(
+          ip,
+          self.date.isoformat(),
```

Comment:
Prefer to work with date objects. Call that only where you need to serialize. Beam will do that for you.

Change the `DateIpKey` to use a date object.

Reply:
Grouping on keys that are not strings is not fully automatic in beam. There's [some trickiness](https://stackoverflow.com/questions/69409693/how-do-i-use-a-datetime-date-value-in-apache-beam-groupby) at least.

Can I attempt this in a seperate change since it's independent of what I'm doing in this PR?

Reply:
This is not a key here. You could convert to string where you extract the key. But, yeah, we don't need to address that in this PR.